mods/boats/init.lua

```diff
@@ -243,7 +243,7 @@ minetest.register_craftitem("boats:boat", {
 		end
 		pointed_thing.under.y = pointed_thing.under.y + 0.5
 		boat = minetest.add_entity(pointed_thing.under, "boats:boat")
-		if boat then
+		if placer and boat then
```

Comment:
Item needs to be removed for nil placers too, so you need to run the itemstack:take_item even if placer is nil.